src/Compilers/CSharp/Portable/Binder/Binder_Attributes.cs

```diff
@@ -217,8 +217,21 @@ private BoundAttribute BindAttributeCore(AttributeSyntax node, NamedTypeSymbol a
 
             analyzedArguments.ConstructorArguments.Free();
 
+            BitVector defaultArguments = default;
+            if (argsToParamsOpt != default && attributeConstructor is not null)
+            {
+                defaultArguments = BitVector.Create(attributeConstructor.ParameterCount);
+                for (int i = 0; i < attributeConstructor.ParameterCount; i++)
+                {
+                    if (!argsToParamsOpt.Contains(i))
```

Comment:
This logic looks incorrect. I think 'argsToParamsOpt' is only provided when arguments are provided out of order of the parameter, using argument labels, for example. 'defaultArguments' indicates which elements of the 'boundConstructorArguments' are synthesized for some optional parameter where an argument wasn't explicitly provided.

It would be great if logic in the below method could be reused here, but I'm not sure how much churn it will end up being or if it will violate assumptions later down the line:

https://github.com/dotnet/roslyn/blob/e37f01d872be3e2809e39e87fe14c24e61a952e7/src/Compilers/CSharp/Portable/Binder/Binder_Invocation.cs#L1271

Reply:
@RikkiGibson I ended up actually calling `BindDefaultArguments`.